src/lib/kclCommands.ts

```diff
@@ -96,6 +104,50 @@ export function kclCommands(commandProps: KclCommandConfig): Command[] {
         }
       },
     },
+    {
+      name: 'Insert',
+      description: 'Insert from a file in the current project directory',
+      icon: 'import',
+      groupId: 'code',
+      hide: DEV || IS_NIGHTLY_OR_DEBUG ? 'web' : 'both',
+      needsReview: true,
+      reviewMessage:
+        'Reminder: point-and-click insert is in development and only supports one part instance per assembly.',
+      args: {
+        path: {
+          inputType: 'options',
+          required: true,
+          options: commandProps.specialPropsForInsertCommand.providedOptions,
+        },
+        localName: {
+          inputType: 'string',
+          required: true,
+        },
+      },
+      onSubmit: (data) => {
+        if (!data) {
+          return new Error('No input provided')
+        }
+
+        const ast = kclManager.ast
+        const { path, localName } = data
+        const { modifiedAst, pathToImportNode, pathToInsertNode } =
+          addImportAndInsert({
+            node: ast,
+            path,
+            localName,
+          })
+        updateModelingState(
+          modifiedAst,
+          EXECUTION_TYPE_REAL,
+          { kclManager, editorManager, codeManager },
+          {
+            skipUpdateAst: true,
+            focusPath: [pathToImportNode, pathToInsertNode],
+          }
+        ).catch(reportRejection)
+      },
+    },
```

Comment:
This started as `insertAstMod` in the modelingMachine file, moved it here after the transition to prefilled paths. Not entirely sure this is the correct way forward.